### PR TITLE
feat: team activity log

### PR DESCRIPTION
**Pull request by pauldambra**
see zendesk feature request https://posthoghelp.zendesk.com/agent/tickets/10303 (moved to PostHog: https://us.posthog.com/project/2/support/tickets/12504)

Adds activity log to the Team API. This will let us capture settings changes like turning session replay on and off and show them in the activity view

Screenshot 2024-02-08 at 21 51 46

I haven't added a describer for everything on the team model it'd take forever. Opting to get this out and we can add describes as we want them

## follow-up

* this would allow an organization view of changes but doesn't try to implement it
* for now this leaves all activity visible only on the team it was logged 
* so for .e.g. we would log deletes but not be able to see them in the UI because you can't visit the deleted team
* let's leave the organization view and its complexity for a follow up

### CODE REVIEW

File: frontend/src/lib/components/ActivityLog/__mocks__/activityLogMocks.ts [hunk 1]
```diff
@@ -2,6 +2,131 @@ import { ActivityLogItem } from 'lib/components/ActivityLog/humanizeActivity'
 
 import { ActivityScope, InsightShortId } from '~/types'
 
+export const teamActivityResponseJson: ActivityLogItem[] = [
+    {
+        user: {
+            first_name: 'sdavasdadadsadas',
```
Comment (marandaneto) [inline frontend/src/lib/components/ActivityLog/__mocks__/activityLogMocks.ts:8]: dfgsdfhdfghfgsdfasdf

**Comment by marandaneto** [comment]
I like the quick turnaround from a support ticket/feature request to a PR adding value, I agree that we can add all the other types of activities as we see them fit, well done.

> are all teams within an org visible to all users in the org? (cos their activity would be)

good question, and probably important to answer before merging, not sure if this could be considered a PII leak or a security breach somehow, if possible, I'd prefer to filter activities by team.

**Comment by daibhin** [comment]
Really cool 👏 Not sure how we communicate it to customers but this feels super valuable as the basis for an audit log that some larger companies require from their vendors. Maybe @fraserhopper knows its worth from his recent foray into compliance

**Comment by fraserhopper** [comment]
This is great! definitely something our bigger, more compliance-y customers will value loads!

**Comment by neilkakkar** [comment]
This is great!

> are all teams within an org visible to all users in the org? (cos their activity would be)
Nope, we can have private projects, (for example, posthog team 2 is private, you're just also in it 😋 ), wouldn't be cool for someone not in it to see activity in there

**Review by daibhin — APPROVED** [review]

**Pull request merged**